chaoslib/litmus/pod_delete/pod-delete.go

```
@@ -29,41 +32,14 @@ func PreparePodDelete(experimentsDetails *experimentTypes.ExperimentDetails, cli
 		log.Infof("[Ramp]: Waiting for the %vs ramp time before injecting chaos", strconv.Itoa(experimentsDetails.RampTime))
 		waitForRampTime(experimentsDetails)
 	}
-
-	// Getting the serviceAccountName
-	err = GetServiceAccount(experimentsDetails, clients)
 	if err != nil {
 		return errors.Errorf("Unable to get the serviceAccountName, err: %v", err)
 	}
 
-	// Generate the run_id
-	runID := GetRunID()
-
-	// Creating a helper pod
-	err = CreateHelperPod(experimentsDetails, clients, runID)
-	if err != nil {
-		return errors.Errorf("Unable to create the helper pod, err: %v", err)
-	}
+	err = PodDeleteChaos(experimentsDetails, clients, eventsDetails, chaosDetails)
 
-	//Checking the status of helper pod
-	log.Info("[Status]: Checking the status of the helper pod")
-	err = status.CheckApplicationStatus(experimentsDetails.ChaosNamespace, "name=pod-delete"+runID, clients)
 	if err != nil {
-		return errors.Errorf("helper pod is not in running state, err: %v", err)
-	}
-
-	// Wait till the completion of helper pod
-	log.Info("[Wait]: waiting till the completion of the helper pod")
-	podStatus, err := status.WaitForCompletion(experimentsDetails.ChaosNamespace, "name=pod-delete"+runID, clients, experimentsDetails.ChaosDuration+experimentsDetails.ChaosInterval+60, "pod-delete")
-	if err != nil || podStatus == "Failed" {
-		return errors.Errorf("helper pod failed due to, err: %v", err)
-	}
-
-	//Deleting the helper pod
-	log.Info("[Cleanup]: Deleting the helper pod")
-	err = DeleteHelperPod(experimentsDetails, clients, runID)
-	if err != nil {
-		return errors.Errorf("Unable to delete the helper pod, err: %v", err)
+		log.Fatalf("Unable to delete the application pods, due to %v", err)
```

Can we return the error here instead of failing it?